analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotTestThisWithIsOperator.CSharp9.cs

```diff
@@ -5,13 +5,33 @@ namespace Tests.Diagnostics
 {
     class Program
     {
+        public int MyProperty { get; set; }
+
+        public int MyProperty2 { get; set; }
+
         public Program(object o)
         {
-            if (this is not IDisposable) // FN
+            if (this is not IDisposable) // Noncompliant {{Offload the code that's conditional on this 'is' test to the appropriate subclass and remove the test.}}
+            {
+            }
+
+            if (o is not IDisposable)
+            {
+            }
+
+            if (this is Program p) // Noncompliant
             {
             }
 
-            switch (this) // FN
+            if (this is var variable) // Noncompliant
+            {
+            }
+
+            if (this is { MyProperty: 5 }) // Noncompliant
```

Comment:
I don't think this is noncompliant. This one doesn't test the type of `this` but uses syntax sugar to test property values.